Spontaneous Error: npm ls -json -prod -long -all failed with code 1 when running serverless deploy or package

```
Error: npm ls -json -prod -long -all failed with code 1
    at ChildProcess.<anonymous> (/Users/mishabruml/****/node_modules/serverless-esbuild/dist/utils.js:53:24)
    at ChildProcess.emit (node:events:390:28)
    at ChildProcess.emit (node:domain:475:12)
    at maybeClose (node:internal/child_process:1064:16)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:301:5)
```

This error started spontaneously with no code or environment changes that i can work out. I have pulled down my master branch, which is working in circleci just fine, but locally I recieve the above error. I am not sure what actually causes this error as the stack trace is unclear. Is it to do with resolving dependencies across monorepo structure? For information, my project looks like: 

```shell
├── package-lock.json
├── package.json
├── packages
│   ├── common
│   │   ├── src
│   │   ├── package.json
│   │   ├── tsconfig.json
│   │   └── tsconfig.lint.json
│   ├── service-a
│   │   ├── esbuild-plugins.js
│   │   ├── package.json
│   │   ├── serverless.yml
│   │   ├── src
│   │   ├── tsconfig.json
│   │   └── tsconfig.lint.json
│   └── service-b
│       ├── package.json
│       ├── src
│       └── tsconfig.json
├── tsconfig.json
└── tsconfig.lint.json
```

Where the `packages` are managed by npm workspaces. It is `service-a` that is the serverless project.

service-a package.json
```json
{
  "engines": {
    "node": "^16.12.0",
    "npm": "^8.1.0"
  },
  "version": "1.0.0",
  "description": "service-a",
  "license": "ISC",
  "scripts": {
    "tsc:check": "tsc -b --listFiles",
    "eslint": "DEBUG=eslint:cli-engine eslint .",
    "prettier": "prettier . --check",
    "test": "jest",
    "sls:deploy": "SLS_DEBUG=* rm -rf .build && sls deploy",
    "sls:package": "rm -rf .build && sls package"
  },
  "devDependencies": {
    "@aws-sdk/types": "^3.38.0",
    "@emarketeer/esbuild-plugin-tsc": "^0.4.1",
    "@tsconfig/node16": "^1.0.2",
    "@types/aws-lambda": "^8.10.84",
    "@types/http-errors": "^1.8.2",
    "@types/jest": "^27.0.2",
    "@types/node": "^14.17.20",
    "esbuild": "^0.14.18",
    "esbuild-plugin-tsc": "^0.3.0",
    "jest": "^27.2.4",
    "serverless": "^3.1.1",
    "serverless-esbuild": "^1.23.4",
    "serverless-offline": "^8.5.0",
    "serverless-step-functions": "^3.5.1",
    "ts-jest": "^27.0.5",
    "ts-node": "^10.4.0",
    "typescript": "^4.5.5"
  },
  "dependencies": {
    "@faker-js/faker": "^6.0.0-alpha.7",
    "@kafkajs/confluent-schema-registry": "^3.2.0",
    "@middy/core": "^2.5.2",
    "@middy/http-json-body-parser": "^2.5.7",
    "@middy/http-response-serializer": "^2.5.7",
    "@middy/ssm": "^2.5.2",
    "@middy/validator": "^2.5.7",
    "@typedorm/common": "^1.14.9",
    "@typedorm/core": "^1.14.9",
    "@types/convict": "^6.1.1",
    "avsc": "^5.7.3",
    "convict": "^6.2.1",
    "http-errors": "^2.0.0",
    "kafkajs": "^1.15.0",
    "middy-zod-validator": "^1.0.5",
    "reflect-metadata": "^0.1.13",
    "uuid": "^8.3.2",
    "zod": "^3.11.6"
  }
}
```

service-a/esbuild-plugins.js
```js
module.exports = [
  require("@emarketeer/esbuild-plugin-tsc")({
    force: true,
  }),
];
```

service-a/serverless.yml
```yml
plugins:
  - serverless-esbuild
  - serverless-step-functions
  - serverless-offline

custom:
  esbuild:
    plugins: esbuild-plugins.js
    external:
      - "@typedorm/core"
      - "@typedorm/common"
    sourcemap: true

package:
  individually: true
```

`npm ls -json -prod -all --depth 0` in `service-a` dir yields: 

```json
{
  "version": "1.0.0",
  "name": "service-a",
  "dependencies": {
    "@faker-js/faker": {
      "version": "6.0.0-alpha.7",
      "resolved": "https://registry.npmjs.org/@faker-js/faker/-/faker-6.0.0-alpha.7.tgz"
    },
    "@kafkajs/confluent-schema-registry": {
      "version": "3.2.1",
      "resolved": "https://registry.npmjs.org/@kafkajs/confluent-schema-registry/-/confluent-schema-registry-3.2.1.tgz"
    },
    "@middy/core": {
      "version": "2.5.7",
      "resolved": "https://registry.npmjs.org/@middy/core/-/core-2.5.7.tgz"
    },
    "@middy/http-json-body-parser": {
      "version": "2.5.7",
      "resolved": "https://registry.npmjs.org/@middy/http-json-body-parser/-/http-json-body-parser-2.5.7.tgz"
    },
    "@middy/http-response-serializer": {
      "version": "2.5.7",
      "resolved": "https://registry.npmjs.org/@middy/http-response-serializer/-/http-response-serializer-2.5.7.tgz"
    },
    "@middy/ssm": {
      "version": "2.5.7",
      "resolved": "https://registry.npmjs.org/@middy/ssm/-/ssm-2.5.7.tgz"
    },
    "@*******/avro-kafkajs": {
      "version": "0.8.1",
      "resolved": "https://registry.npmjs.org/@*******/avro-kafkajs/-/avro-kafkajs-0.8.1.tgz"
    },
    "@*******/castle": {
      "version": "0.8.1",
      "resolved": "https://registry.npmjs.org/@*******/castle/-/castle-0.8.1.tgz"
    },
    "@sinclair/typebox": {
      "version": "0.23.4",
      "resolved": "https://registry.npmjs.org/@sinclair/typebox/-/typebox-0.23.4.tgz"
    },
    "@typedorm/common": {
      "version": "1.14.9",
      "resolved": "https://registry.npmjs.org/@typedorm/common/-/common-1.14.9.tgz"
    },
    "@typedorm/core": {
      "version": "1.14.9",
      "resolved": "https://registry.npmjs.org/@typedorm/core/-/core-1.14.9.tgz"
    },
    "@types/convict": {
      "version": "6.1.1",
      "resolved": "https://registry.npmjs.org/@types/convict/-/convict-6.1.1.tgz"
    },
    "avsc": {
      "version": "5.7.3",
      "resolved": "https://registry.npmjs.org/avsc/-/avsc-5.7.3.tgz"
    },
    "convict": {
      "version": "6.2.1",
      "resolved": "https://registry.npmjs.org/convict/-/convict-6.2.1.tgz"
    },
    "http-errors": {
      "version": "2.0.0",
      "resolved": "https://registry.npmjs.org/http-errors/-/http-errors-2.0.0.tgz"
    },
    "kafkajs": {
      "version": "1.16.0",
      "resolved": "https://registry.npmjs.org/kafkajs/-/kafkajs-1.16.0.tgz"
    },
    "reflect-metadata": {
      "version": "0.1.13",
      "resolved": "https://registry.npmjs.org/reflect-metadata/-/reflect-metadata-0.1.13.tgz"
    },
    "uuid": {
      "version": "8.3.2",
      "resolved": "https://registry.npmjs.org/uuid/-/uuid-8.3.2.tgz"
    },
    "zod": {
      "version": "3.13.2",
      "resolved": "https://registry.npmjs.org/zod/-/zod-3.13.2.tgz"
    }
  }
}
```

Thanks for looking into this. Here is the output. Some redaction due to private repos 😄 

```shell

> sls:deploy
> SLS_DEBUG=* rm -rf .build && sls deploy

8.1.0
 
{
  "version": "1.0.0",
  "name": "service-a",
  "engines": {
    "node": "^16.12.0",
    "npm": "^8.1.0"
  },
  "description": "service-a",
  "author": "*****",
  "license": "ISC",
  "scripts": {
    "tsc:check": "tsc -b --listFiles",
    "eslint": "DEBUG=eslint:cli-engine eslint .",
    "prettier": "prettier . --check",
    "test": "jest",
    "sls:deploy": "SLS_DEBUG=* rm -rf .build && sls deploy",
    "sls:package": "rm -rf .build && sls package"
  },
  "devDependencies": {
    "@aws-sdk/types": "^3.38.0",
    "@emarketeer/esbuild-plugin-tsc": "^0.4.1",
    "@tsconfig/node16": "^1.0.2",
    "@types/aws-lambda": "^8.10.84",
    "@types/http-errors": "^1.8.2",
    "@types/jest": "^27.0.2",
    "@types/node": "^14.17.20",
    "esbuild": "^0.14.18",
    "esbuild-plugin-tsc": "^0.3.0",
    "jest": "^27.2.4",
    "serverless": "^3.1.1",
    "serverless-esbuild": "^1.23.4",
    "serverless-iam-roles-per-function": "^3.2.0",
    "serverless-offline": "^8.5.0",
    "serverless-step-functions": "^3.5.1",
    "ts-jest": "^27.0.5",
    "ts-node": "^10.4.0"
  },
  "extraneous": false,
  "path": "/Users/mishabruml/*****/**********/packages/service-a",
  "_dependencies": {
    "@faker-js/faker": "^6.0.0-alpha.7",
    "@kafkajs/confluent-schema-registry": "^3.2.0",
    "@middy/core": "^2.5.2",
    "@middy/http-json-body-parser": "^2.5.7",
    "@middy/http-response-serializer": "^2.5.7",
    "@middy/ssm": "^2.5.2",
    "@*****/avro-kafkajs": "^0.8.1",
    "@*****/castle": "^0.8.1",
    "@sinclair/typebox": "^0.23.3",
    "@types/convict": "^6.1.1",
    "avsc": "^5.7.3",
    "convict": "^6.2.1",
    "http-errors": "^2.0.0",
    "kafkajs": "^1.15.0",
    "reflect-metadata": "^0.1.13",
    "uuid": "^8.3.2",
    "zod": "^3.11.6"
  },
  "peerDependencies": {},
  "problems": [
    "missing: @faker-js/faker@^6.0.0-alpha.7, required by service-a@1.0.0",
    "missing: @kafkajs/confluent-schema-registry@^3.2.0, required by service-a@1.0.0",
    "missing: @middy/core@^2.5.2, required by service-a@1.0.0",
    "missing: @middy/http-json-body-parser@^2.5.7, required by service-a@1.0.0",
    "missing: @middy/http-response-serializer@^2.5.7, required by service-a@1.0.0",
    "missing: @middy/ssm@^2.5.2, required by service-a@1.0.0",
    "missing: @*****/avro-kafkajs@^0.8.1, required by service-a@1.0.0",
    "missing: @*****/castle@^0.8.1, required by service-a@1.0.0",
    "missing: @sinclair/typebox@^0.23.3, required by service-a@1.0.0",
    "missing: @types/convict@^6.1.1, required by service-a@1.0.0",
    "missing: avsc@^5.7.3, required by service-a@1.0.0",
    "missing: convict@^6.2.1, required by service-a@1.0.0",
    "missing: http-errors@^2.0.0, required by service-a@1.0.0",
    "missing: kafkajs@^1.15.0, required by service-a@1.0.0",
    "missing: reflect-metadata@^0.1.13, required by service-a@1.0.0",
    "missing: uuid@^8.3.2, required by service-a@1.0.0",
    "missing: zod@^3.11.6, required by service-a@1.0.0"
  ],
  "dependencies": {
    "@faker-js/faker": {
      "required": "^6.0.0-alpha.7",
      "missing": true,
      "problems": [
        "missing: @faker-js/faker@^6.0.0-alpha.7, required by service-a@1.0.0"
      ]
    },
    "@kafkajs/confluent-schema-registry": {
      "missing": true,
      "problems": [
        "missing: @kafkajs/confluent-schema-registry@^3.2.0, required by service-a@1.0.0"
      ]
    },
    "@middy/core": {
      "missing": true,
      "problems": [
        "missing: @middy/core@^2.5.2, required by service-a@1.0.0"
      ]
    },
    "@middy/http-json-body-parser": {
      "missing": true,
      "problems": [
        "missing: @middy/http-json-body-parser@^2.5.7, required by service-a@1.0.0"
      ]
    },
    "@middy/http-response-serializer": {
      "missing": true,
      "problems": [
        "missing: @middy/http-response-serializer@^2.5.7, required by service-a@1.0.0"
      ]
    },
    "@middy/ssm": {
      "missing": true,
      "problems": [
        "missing: @middy/ssm@^2.5.2, required by service-a@1.0.0"
      ]
    },
    "@*****/avro-kafkajs": {
      "missing": true,
      "problems": [
        "missing: @*****/avro-kafkajs@^0.8.1, required by service-a@1.0.0"
      ]
    },
    "@*****/castle": {
      "missing": true,
      "problems": [
        "missing: @*****/castle@^0.8.1, required by service-a@1.0.0"
      ]
    },
    "@sinclair/typebox": {
      "missing": true,
      "problems": [
        "missing: @sinclair/typebox@^0.23.3, required by service-a@1.0.0"
      ]
    },
    "@types/convict": {
      "missing": true,
      "problems": [
        "missing: @types/convict@^6.1.1, required by service-a@1.0.0"
      ]
    },
    "avsc": {
      "missing": true,
      "problems": [
        "missing: avsc@^5.7.3, required by service-a@1.0.0"
      ]
    },
    "convict": {
      "missing": true,
      "problems": [
        "missing: convict@^6.2.1, required by service-a@1.0.0"
      ]
    },
    "http-errors": {
      "missing": true,
      "problems": [
        "missing: http-errors@^2.0.0, required by service-a@1.0.0"
      ]
    },
    "kafkajs": {
      "missing": true,
      "problems": [
        "missing: kafkajs@^1.15.0, required by service-a@1.0.0"
      ]
    },
    "reflect-metadata": {
      "missing": true,
      "problems": [
        "missing: reflect-metadata@^0.1.13, required by service-a@1.0.0"
      ]
    },
    "uuid": {
      "missing": true,
      "problems": [
        "missing: uuid@^8.3.2, required by service-a@1.0.0"
      ]
    },
    "zod": {
      "missing": true,
      "problems": [
        "missing: zod@^3.11.6, required by service-a@1.0.0"
      ]
    }
  }
}
 npm ERR! code ELSPROBLEMS
npm ERR! missing: @faker-js/faker@^6.0.0-alpha.7, required by service-a@1.0.0
npm ERR! missing: @kafkajs/confluent-schema-registry@^3.2.0, required by service-a@1.0.0
npm ERR! missing: @middy/core@^2.5.2, required by service-a@1.0.0
npm ERR! missing: @middy/http-json-body-parser@^2.5.7, required by service-a@1.0.0
npm ERR! missing: @middy/http-response-serializer@^2.5.7, required by service-a@1.0.0
npm ERR! missing: @middy/ssm@^2.5.2, required by service-a@1.0.0
npm ERR! missing: @*****/avro-kafkajs@^0.8.1, required by service-a@1.0.0
npm ERR! missing: @*****/castle@^0.8.1, required by service-a@1.0.0
npm ERR! missing: @sinclair/typebox@^0.23.3, required by service-a@1.0.0
npm ERR! missing: @types/convict@^6.1.1, required by service-a@1.0.0
npm ERR! missing: avsc@^5.7.3, required by service-a@1.0.0
npm ERR! missing: convict@^6.2.1, required by service-a@1.0.0
npm ERR! missing: http-errors@^2.0.0, required by service-a@1.0.0
npm ERR! missing: kafkajs@^1.15.0, required by service-a@1.0.0
npm ERR! missing: reflect-metadata@^0.1.13, required by service-a@1.0.0
npm ERR! missing: uuid@^8.3.2, required by service-a@1.0.0
npm ERR! missing: zod@^3.11.6, required by service-a@1.0.0
{
  "error": {
    "code": "ELSPROBLEMS",
    "summary": "missing: @faker-js/faker@^6.0.0-alpha.7, required by service-a@1.0.0\nmissing: @kafkajs/confluent-schema-registry@^3.2.0, required by service-a@1.0.0\nmissing: @middy/core@^2.5.2, required by service-a@1.0.0\nmissing: @middy/http-json-body-parser@^2.5.7, required by service-a@1.0.0\nmissing: @middy/http-response-serializer@^2.5.7, required by service-a@1.0.0\nmissing: @middy/ssm@^2.5.2, required by service-a@1.0.0\nmissing: @*****/avro-kafkajs@^0.8.1, required by service-a@1.0.0\nmissing: @*****/castle@^0.8.1, required by service-a@1.0.0\nmissing: @sinclair/typebox@^0.23.3, required by service-a@1.0.0\nmissing: @types/convict@^6.1.1, required by service-a@1.0.0\nmissing: avsc@^5.7.3, required by service-a@1.0.0\nmissing: convict@^6.2.1, required by service-a@1.0.0\nmissing: http-errors@^2.0.0, required by service-a@1.0.0\nmissing: kafkajs@^1.15.0, required by service-a@1.0.0\nmissing: reflect-metadata@^0.1.13, required by service-a@1.0.0\nmissing: uuid@^8.3.2, required by service-a@1.0.0\nmissing: zod@^3.11.6, required by service-a@1.0.0",
    "detail": ""
  }
}

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/mishabruml/.npm/_logs/2022-03-03T14_33_14_772Z-debug.log

Error:
Error: npm ls -json -prod -long -all failed with code 1
    at ChildProcess.<anonymous> (/Users/mishabruml/*****/**********/node_modules/serverless-esbuild/dist/utils.js:54:24)
    at ChildProcess.emit (node:events:390:28)
    at ChildProcess.emit (node:domain:475:12)
    at maybeClose (node:internal/child_process:1064:16)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:301:5)
```

--- a/src/packagers/npm.ts
+++ b/src/packagers/npm.ts
@@ -1,4 +1,4 @@
-import { any, isEmpty, reduce, replace, split, startsWith } from 'ramda';
+import { any, isEmpty, replace, split, startsWith, takeWhile } from 'ramda';
 import * as path from 'path';
 
 import { DependenciesResult, DependencyMap, JSONObject } from '../types';
@@ -125,6 +125,7 @@ export class NPM implements Packager {
       { npmError: 'extraneous', log: false },
       { npmError: 'missing', log: false },
       { npmError: 'peer dep missing', log: true },
+      { npmError: 'code ELSPROBLEMS', log: false },
     ];
 
     let parsedDeps: NpmV6Deps | NpmV7Deps;
@@ -134,25 +135,21 @@ export class NPM implements Packager {
     } catch (err) {
       if (err instanceof SpawnError) {
         // Only exit with an error if we have critical npm errors for 2nd level inside
-        const errors = split('\n', err.stderr);
-        const failed = reduce(
-          (f, error) => {
-            if (f) {
-              return true;
-            }
-            return (
-              !isEmpty(error) &&
-              !any(
-                (ignoredError) => startsWith(`npm ERR! ${ignoredError.npmError}`, error),
-                ignoredNpmErrors
-              )
-            );
-          },
-          false,
-          errors
+        // Split the stderr by \n character to get the npm ERR! plaintext lines, ignore additonal JSON blob (emitted by npm >=7)
+        // see https://github.com/serverless-heaven/serverless-webpack/pull/782 and https://github.com/floydspace/serverless-esbuild/issues/288
+        const lines = split('\n', err.stderr);
+        const npmErrors = takeWhile((line) => line !== '{', lines);
+
+        const hasThrowableErrors = npmErrors.every(
+          (error) =>
+            !isEmpty(error) &&
+            !any(
+              (ignoredError) => startsWith(`npm ERR! ${ignoredError.npmError}`, error),
+              ignoredNpmErrors
+            )
         );
 
-        if (!failed && !isEmpty(err.stdout)) {
+        if (!hasThrowableErrors && !isEmpty(err.stdout)) {
           return { stdout: err.stdout };
         }
       }

--- a/src/tests/packagers/npm.test.ts
+++ b/src/tests/packagers/npm.test.ts
@@ -1004,14 +1004,6 @@ describe('NPM Packager', () => {
     expect(v7dependencies).toStrictEqual(expectedResult);
   });
 
-  test.each([
-    { a: 1, b: 1, expected: 2 },
-    { a: 1, b: 2, expected: 3 },
-    { a: 2, b: 1, expected: 3 },
-  ])('.add($a, $b)', ({ a, b, expected }) => {
-    expect(a + b).toBe(expected);
-  });
-
   it.each([
     'npm ERR! code ELSPROBLEMS',
     'npm ERR! extraneous: foo@1.2.3 ./bar/node_modules/foo',

--- a/src/tests/packagers/npm.test.ts
+++ b/src/tests/packagers/npm.test.ts
@@ -1003,4 +1003,123 @@ describe('NPM Packager', () => {
     expect(v6dependencies).toStrictEqual(expectedResult);
     expect(v7dependencies).toStrictEqual(expectedResult);
   });
+
+  test.each([
+    { a: 1, b: 1, expected: 2 },
+    { a: 1, b: 2, expected: 3 },
+    { a: 2, b: 1, expected: 3 },
+  ])('.add($a, $b)', ({ a, b, expected }) => {
+    expect(a + b).toBe(expected);
+  });
+
+  it.each([
+    'npm ERR! code ELSPROBLEMS',
+    'npm ERR! extraneous: foo@1.2.3 ./bar/node_modules/foo',
+    'npm ERR! missing: foo-1@1.0.0, required by foo-2@1.0.0',
+    'npm ERR! peer dep missing: foo@1.2.3',
+  ])('should ignore npm error "%s"', async (ignoredNpmError) => {
+    const v7depsList: NpmV7Deps = {
+      version: '1.0.0',
+      name: 'serverless-example',
+      description: 'Packaged externals for serverless-example',
+      private: true,
+      scripts: {},
+      _id: 'serverless-example@1.0.0',
+      extraneous: false,
+      path: '/workdir/.esbuild/.build',
+      _dependencies: {
+        'samchungy-a': '2.0.0',
+        'samchungy-b': '2.0.0',
+      },
+      devDependencies: {},
+      peerDependencies: {},
+      dependencies: {
+        'samchungy-a': {
+          version: '2.0.0',
+          resolved: 'https://registry.npmjs.org/samchungy-a/-/samchungy-a-2.0.0.tgz',
+          name: 'samchungy-a',
+          integrity:
+            'sha512-gUv/cvd9AFYvvGep0e9m1wSAf3dfnb71eri5TjtgC6N7qvJALXFaFVOkLNBHEYGEm2ZJdosXvGqr3ISZ7Yh46Q==',
+          _id: 'samchungy-a@2.0.0',
+          extraneous: false,
+          path: '/workdir/.esbuild/.build/node_modules/samchungy-a',
+          _dependencies: {
+            'samchungy-dep-a': '1.0.0',
+          },
+          devDependencies: {},
+          peerDependencies: {},
+          dependencies: {
+            'samchungy-dep-a': {
+              version: '1.0.0',
+              resolved: 'https://registry.npmjs.org/samchungy-dep-a/-/samchungy-dep-a-1.0.0.tgz',
+              name: 'samchungy-dep-a',
+              integrity:
+                'sha512-NVac5aAU+p7bsIrUTQO438vAO8MHyNILbeckhzxhadIUqGx3L9kEZ5HTqZ+XqDIRARmOU6UmFtus6Bc7q5+mWA==',
+              _id: 'samchungy-dep-a@1.0.0',
+              extraneous: false,
+              path: '/workdir/.esbuild/.build/node_modules/samchungy-dep-a',
+              _dependencies: {},
+              devDependencies: {},
+              peerDependencies: {},
+            },
+          },
+        },
+        'samchungy-b': {
+          version: '2.0.0',
+          resolved: 'https://registry.npmjs.org/samchungy-b/-/samchungy-b-2.0.0.tgz',
+          name: 'samchungy-b',
+          integrity:
+            'sha512-i42OG9FC2Py3RfbI8bBFZi3VoN7+MxM0OUvFcWrsIgqvZMUDVI4hNKHqpE6GTt07gDDqQnxlMNehbrsQLtHRVA==',
+          _id: 'samchungy-b@2.0.0',
+          extraneous: false,
+          path: '/workdir/.esbuild/.build/node_modules/samchungy-b',
+          _dependencies: {
+            'samchungy-dep-a': '2.0.0',
+          },
+          devDependencies: {},
+          peerDependencies: {},
+          dependencies: {
+            'samchungy-dep-a': {
+              version: '2.0.0',
+              resolved: 'https://registry.npmjs.org/samchungy-dep-a/-/samchungy-dep-a-2.0.0.tgz',
+              name: 'samchungy-dep-a',
+              integrity:
+                'sha512-Yp30ASjwmyLWCGhlLTqWZa8MlBeBiaaHsmxXMwwQxK/o044vhCsPeugHqhtsZq7Xiq68/TcBux/LKId6eyPNjA==',
+              _id: 'samchungy-dep-a@2.0.0',
+              extraneous: false,
+              path: '/workdir/.esbuild/.build/node_modules/samchungy-b/node_modules/samchungy-dep-a',
+              _dependencies: {},
+              devDependencies: {},
+              peerDependencies: {},
+            },
+          },
+        },
+      },
+    };
+
+    const npm7stderr: string = ''.concat(
+      ignoredNpmError,
+      '\n',
+      JSON.stringify(
+        {
+          error: {
+            code: 'TEST_ERROR_CODE',
+            summary: 'test summary',
+            detail: 'test detail',
+          },
+        },
+        null,
+        1
+      )
+    );
+
+    spawnSpy
+      .mockResolvedValueOnce({ stderr: '', stdout: '7.0.0' })
+      .mockRejectedValueOnce(
+        new utils.SpawnError('a spawn error', JSON.stringify(v7depsList), npm7stderr)
+      );
+
+    const result = await npm.getProdDependencies(path);
+    expect(result).toStrictEqual({ stdout: JSON.stringify(v7depsList) });
+  });
 });

